### PR TITLE
Cleanup package architecture

### PR DESCRIPTION
**Pull request by sydneyli**
This will probably conflict with any changes, but the gist of it is to move things and clean up the API abstraction a bit!

`api.go`, `middleware.go`, `email.go` to `api/` and `util.go` to `util/`-- although `util` is bare enough that I might move the individual functions to other places.

There's no rush to land this, just something I'm playing around with. Let me know what you think!

### CODE REVIEW

**Comment by vbrown608** [comment]
I like this direction! I really like how everything in the `api` package is now getting automatically namespaced. In the past I've wondered if each handler should get its own file in an `api` package, along with any associated helpers, but that could just be my indoctrination with ruby/rails talking.

What's the logic behind including `email.go` in that package? It seems a bit different from the rest, which is mostly handlers and code to support them.

**Comment by sydneyli** [comment]
> In the past I've wondered if each handler should get its own file in an api package, along with any associated helpers, but that could just be my indoctrination with ruby/rails talking.

There's a lot of different logic in api.go, and I was thinking it does at least make sense to separate the web-request-handling logic from the others.

> What's the logic behind including email.go in that package? It seems a bit different from the rest, which is mostly handlers and code to support them.

Good question! I had it in a separate package previously, then decided it really didn't need to be there yet since `api` was (currently) the only consumer of the `email` package. I expect that might change, so it might be good to do that pre-emptively.

**Comment by sydneyli** [comment]
I'm still not certain the current separation between `email` and `api` is great, but I think it's better than before. Might try and clean it up after!

File: main.go [hunk 1]
```diff
@@ -10,39 +10,24 @@ import (
...
 	"github.com/EFForg/starttls-backend/policy"
 	"github.com/EFForg/starttls-backend/stats"
+	"github.com/EFForg/starttls-backend/util"
 	"github.com/EFForg/starttls-backend/validator"
 
 	"github.com/getsentry/raven-go"
 	_ "github.com/joho/godotenv/autoload"
 )
 
-func pingHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (vbrown608) [inline main.go]: Nice to get this out of main.go :)

**Pull request merged**